pedsnetdcc/check_fact_relationship.py

```diff
@@ -0,0 +1,283 @@
+SELECT COUNT(*) FROM fact_relationship WHERE domain_concept_id_1 = 8
+'''
+
+# The below bad fact_relationship record count statements return the
+# number of bad *records* not the number of bad *references*. There may be
+# more than one (two) bad references per record.
+
+sql_bad_obs_count = '''
+SELECT COUNT(*) FROM (
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN observation o ON f.fact_id_1 = o.observation_id
+    WHERE domain_concept_id_1 = 27 AND o.observation_id IS NULL
+UNION
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN observation o ON f.fact_id_2 = o.observation_id
+    WHERE domain_concept_id_2 = 27 AND o.observation_id IS NULL
+) q
+'''
+
+sql_bad_meas_count = '''
+SELECT COUNT(*) FROM (
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN measurement m ON f.fact_id_1 = m.measurement_id
+    WHERE domain_concept_id_1 = 21 AND m.measurement_id IS NULL
+UNION
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN measurement m ON f.fact_id_2 = m.measurement_id
+    WHERE domain_concept_id_2 = 21 AND m.measurement_id IS NULL
+) q
+'''
+
+sql_bad_visit_count = '''
+SELECT COUNT(*) FROM (
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN visit_occurrence v ON f.fact_id_1 = v.visit_occurrence_id
+    WHERE domain_concept_id_1 = 8 AND v.visit_occurrence_id IS NULL
+UNION
+    SELECT f.*
+    FROM fact_relationship f
+        LEFT JOIN visit_occurrence v ON f.fact_id_2 = v.visit_occurrence_id
+    WHERE domain_concept_id_2 = 8 AND v.visit_occurrence_id IS NULL
+) q
+'''
+
+sql_bad_obs_1_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN observation o ON f.fact_id_1 = o.observation_id
+WHERE domain_concept_id_1 = 27 AND o.observation_id IS NULL
+LIMIT 1
+'''
+
+sql_bad_obs_2_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN observation o ON f.fact_id_2 = o.observation_id
+WHERE domain_concept_id_2 = 27 AND o.observation_id IS NULL
+LIMIT 1
+'''
+
+sql_bad_meas_1_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN measurement m ON f.fact_id_1 = m.measurement_id
+WHERE domain_concept_id_1 = 21 AND m.measurement_id IS NULL
+LIMIT 1
+'''
+
+sql_bad_meas_2_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN measurement m ON f.fact_id_2 = m.measurement_id
+WHERE domain_concept_id_2 = 21 AND m.measurement_id IS NULL
+LIMIT 1
+'''
+
+sql_bad_visit_1_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN visit_occurrence v ON f.fact_id_1 = v.visit_occurrence_id
+WHERE domain_concept_id_1 = 8 AND v.visit_occurrence_id IS NULL
+LIMIT 1
+'''
+
+sql_bad_visit_2_sample = '''
+SELECT f.*
+FROM fact_relationship f
+    LEFT JOIN visit_occurrence v ON f.fact_id_2 = v.visit_occurrence_id
+WHERE domain_concept_id_2 = 8 AND v.visit_occurrence_id IS NULL
+LIMIT 1
+'''
+
+logger = logging.getLogger(__name__)
+
+
+def check_fact_relationship(conn_str, output='percent'):
+
+    if output == 'percent':
```

Comment:
Analogous to `sync_observation_period.py`, should this emit an info-level "starting" log message?